### ISSUE TITLE
display nested "Known Religious Communities"

### ISSUE THREAD

**Issue by tacarlson** [srophe/srp-places-app#117]
The list of confession attestations should be nested according to the hierarchy developed by tacarlson in srophe/places#5  How to do this will require further discussion.

**Comment by tacarlson** [comment]
I added attestations to https://github.com/srophe/places/blob/master/xml/78.xml and mocked up how the confessions section might be displayed at https://github.com/srophe/places/blob/master/xml/Examples/78.html (a very rough mockup).  Note how the .html sorts the attestations by start date, and the date information under the Known Religious Communities gives a link from each boundary date to the relevant attestation.  The html also needs to generate as much of the hierarchy from http://srophe.github.io/srp-places-app/help/confessions.html as is populated with either attestations or with footnotes, and dates for attestations of children confessions get populated upward to dates for attestations of parent confessions.  Note how the .html ignores the dates on the `<state type="confession">` (those dates will be used for searching, but will not be displayed to human users.  Also note the change in the disclaimer under Known Religious Communities.

If a Known Religious Community has no attestation (but has a footnote), then the "date" text should read "(no attestations yet recorded)".  If a Known Religious Community has a single attestation in the date YYYY, then the date text should read "(attested YYYY)".  If a Known Religious Community has two or more attestations, with the earliest having the date YYYY and the latest having the date ZZZZ, then the date text should read "(attested as early as YYYY, as late as ZZZZ)".

But attestations are not necessarily precisely dated, so if an attestation has either a `@notBefore` or a `@notAfter` attribute, then we presume the window is relatively small and stick "around" before the value of the attribute.  If an attestation has both a `@notBefore` and a `@notAfter` attribute, then in place of the single date should read "between XXXX and YYYY".

Putting this together, assume for the sake of argument the confession "Muslim" has two attestations, one with a `@notBefore="0641"` and one dated with `@notBefore="1925" @notAfter="1930"`, then the text in the Known Religious Communities section (excluding the links to the relevant attestation) should read: "Muslim (attested as early as around 641, as late as between 1925 and 1930)".

@davidamichelson : Is thiswhat we want?  I think this is what we discussed on Wednesday.
@wsalesky : does this look computationally possible?  
Let's discuss this coming Wednesday before implementing.

**Comment by wsalesky** [comment]
So. I have been spending a lot of time with this data. I have a question. I have the "tree" populating correctly, so when you have Syrian Orthodox, you also see West Syrian and Syriac even though they are not referenced by the place data. However, I can not generate any dates for these because there is no reference to them in the data. 

So the data looks like this (still ironing out kinks in date display):
Christians (attested as early as 201, as late as 1624)
    Syriac 
       West Syrian
             Melkite (attested as early as 522 attested as early as 540)
             Syrian Orthodox (attested as early as 1146)
    Latin (no attestations yet recorded)
Jews (no attestations yet recorded)
Muslims (no attestations yet recorded)

Is this okay, do you want to have the dates in the confessions.xml instead so that they can be pulled with the confession data? Hope that makes sense. 
-Winona

**Comment by wsalesky** [comment]
Sorry, the list did not display properly in the comment, but all the subsections are properly indented.

**Comment by tacarlson** [comment]
I noticed from this display that I had linked two attestations to the wrong confession!  So I have just fixed the 78.xml data.  Thanks for visualizing this!

I was wondering whether the dates could be populated upward in the list as well?  So for West Syrian, it's "attested as early as" date would be the earliest attestation of itself or of any of its children, and similar for Syriac and for Christian.  Mutatis mutandis, the "attested as late as" date for West Syrian would be the latest attestation of itself or of any of its children.  For any of these, there could be attestations at that level (e.g. the archaeological remains of a church testify to a Christian presence, but perhaps not what kind of church), and there could be attestations of child nodes.  Is it possible to populate these dates upward?

I also noticed the date text on "Melkite" reads "attested as early as 522 attested as early as 540" rather than "attested as early as 522, as late as around 540".

Thanks for all your work!
